tests/manage/test_create_storage_class_pvc.py

```diff
@@ -1,135 +1,133 @@
 import logging
-
 import pytest
 
-from ocs_ci.ocs import ocp, constants
-from ocs_ci.framework import config
+from ocs_ci.ocs import constants
 from ocs_ci.framework.testlib import tier1, ManageTest
-from ocs_ci.ocs.resources.ocs import OCS
-from ocs_ci.ocs.resources.pod import get_admin_key_from_ceph_tools
-from ocs_ci.ocs.resources.pvc import PVC
 from tests import helpers
-from ocs_ci.utility import templating
 
 log = logging.getLogger(__name__)
 
 
-POD = ocp.OCP(kind='Pod', namespace=config.ENV_DATA['cluster_namespace'])
-CEPH_OBJ = None
+@pytest.fixture(scope='function')
+def test_fixture_rbd(request):
+    def finalizer():
+        teardown_rbd()
+
+    request.addfinalizer(finalizer)
+    setup_rbd()
 
 
-@pytest.fixture()
-def test_fixture(request):
+def setup_rbd():
     """
-    Create disks
+    Setting up the environment
+    Creating replicated pool,secret,storageclass for rbd
     """
-    self = request.node.cls
-    setup_fs(self)
-    yield
-    teardown_fs()
+    log.info("Creating CephBlockPool")
+    global RBD_POOL
+    RBD_POOL = helpers.create_ceph_block_pool()
+    global RBD_SECRET_OBJ
+    RBD_SECRET_OBJ = helpers.create_secret(constants.CEPHBLOCKPOOL)
+    global RBD_SC_OBJ
+    log.info("Creating RBD Storage class ")
+    RBD_SC_OBJ = helpers.create_storage_class(
+        interface_type=constants.CEPHBLOCKPOOL,
+        interface_name=RBD_POOL.name,
+        secret_name=RBD_SECRET_OBJ.name
+    )
 
 
-def setup_fs(self):
+def teardown_rbd():
     """
-    Setting up the environment for the test
+    Tearing down the environment
+    Deleting pod,replicated pool,pvc,storageclass,secret of rbd
     """
-    global CEPH_OBJ
-    self.fs_data = templating.load_yaml_to_dict(constants.CEPHFILESYSTEM_YAML)
-    self.fs_data['metadata']['name'] = helpers.create_unique_resource_name(
-        'test', 'cephfs'
-    )
-    self.fs_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
-    CEPH_OBJ = OCS(**self.fs_data)
-    CEPH_OBJ.create()
-    assert POD.wait_for_resource(
-        condition='Running', selector='app=rook-ceph-mds'
+    global RBD_PVC_OBJ, RBD_POD_OBJ, RBD_PV
+    log.info('deleting rbd pod')
+    RBD_POD_OBJ.delete()
+    log.info("Deleting RBD PVC")
+    RBD_PVC_OBJ.delete()
+    assert helpers.validate_pv_delete(RBD_PVC_OBJ.backed_pv)
+    log.info("Deleting CEPH BLOCK POOL")
+    RBD_POOL.delete()
+    log.info("Deleting RBD Secret")
+    RBD_SECRET_OBJ.delete()
+    log.info("Deleting RBD Storageclass")
+    RBD_SC_OBJ.delete()
+
+
+@pytest.fixture(scope='function')
+def test_fixture_cephfs(request):
+    def finalizer():
+        teardown_fs()
+
+    request.addfinalizer(finalizer)
```

Comment:
Same here, add the finalizer directly.
```suggestion
    request.addfinalizer(teardown_fs)
```

Reply:
Thanks, added in the script